### DISCUSSION TITLE
Would it be helpful to have the ability to directly create tables and figures in the template and have the option to call pre-generated ones as well?

### DISCUSSION THREAD

**Discussion by Schiano-NOAA** [nmfs-ost/asar#37] [Q&A]
Currently, the generated template includes a child document for both tables and figures. When first rendered in full, the document will generate the tables and figures written in those quarto files; however, there could be a possibility that the analyst may want to use progenerated tables and figures. In this case, we could override these files and tell R to read in the special tables. 

In the alternative situation, alternative text will not be associated with the images (unless it is possible to embed this information with the image). I am also thinking of setting alternative text as a separate function instead of programming directly in the template.

Let me know your thoughts!

**Comment by kellijohnson-NOAA** [comment]
I do not think that there should be any built in functionality right now to use things that are not the default. Users will customize it to their needs but more work should be on standardizing things across centers so we can work towards shared code rather than the continued division between centers. The easier that you make it to switch stuff in and out then the more people will choose that route.

Regarding alternative text. I am a big fan of modularity but it would be hard to know what the alternative text should be if the function does not have access to the data. I am wondering if the automated alternative text is going to be specific to each figure or boiler plate information on where to find the raw data? If the latter, then I think that it would be a stand-a-lone function that takes very few inputs and provides a sentence based on those inputs where the reader can find the raw data, such as in section 5 of this json file or filter this data frame by module == "spawning biomass".
